Sources/HTTPClient/HTTPError.swift

```diff
@@ -1,10 +1,27 @@
 import Foundation
 
-enum HTTPError: Error {
+public enum HTTPError: Error, Equatable {
     case malformedURL(HTTPService)
+    case couldNotConnect
     case dataTaskError(Error)
     case noResponse
     case badStatusCode(Int)
     case responseDataNil
     case jsonDecodeFailed(Error, Data)
+    
+    public static func ==(lhs: HTTPError, rhs: HTTPError) -> Bool {
```

Comment:
nit: I wonder if you could use `if case let` pattern matching instead of conforming to Equatable. When I'm not sure how to write it I use this website https://fuckingifcaseletsyntax.com/ :D

Reply:
Not necessarily `if case let` works in same way as `switch`, and you don't need Equatable for the switch.